### ISSUE TITLE
Configuration options for curl arguments

### ISSUE THREAD

**Issue by trws** [spack/spack#69]
The sourceforge download issue, #59, turned out to be due to an overridden user agent.  Having a way to inject options into the curl invocation to avoid or adjust things like this in the user's environment would be useful.

**Comment by mathstuf** [comment]
The `-q` flag can be passed to curl to avoid reading a `.curlrc` file (must be the first argument).

**Comment by tgamblin** [comment]
@mathstuf: This might be a good thing to add to Spack as a config option.  Some people need to curl through a proxy, though, so i think it's actually unclear what the default curl settings should be.  Some users probably expect Spack to use the settings in `.curlrc`

**Comment by mathstuf** [comment]
For proxies, there's the `http_proxy` environment variable which should be obeyed (apparently `HTTP_PROXY`, `ALL_PROXY`, and `${proto}_PROXY` are also all valid for curl).

**Comment by adamjstewart** [comment]
@tgamblin Can this be added to `config.yaml`?

**Comment by alalazo** [comment]
@tgamblin Is this (injecting options into `curl` from config) still something we want?

**Comment by joequant** [comment]
Yes it is.  I'm working on the edm4hep project and we are building an entire distribution from scratch.  Proxy servers would be very useful.

**Comment by becker33** [comment]
Now that curl is not the default fetch method for Spack, can we fold this into the config options by which you tell spack to use curl? e.g. `spack config add config:url_fetch_method:'curl -q'`?

**Comment by trws** [comment]
Wow... This issue number has _two_ digits... just a hair under a decade ago. Do we have active issues that are older?

I don't see why not @becker33, probably pretty straightforward that way too, either like you show or as an array that starts with the name of the method.

**Comment by trws** [comment]
Just for my personal amusement, we do not in fact have older open issues. This is the oldest by over 900 issue numbers and almost a year:

![Image](https://github.com/user-attachments/assets/fc61adc2-05fc-4479-909e-c475873bfa43)

Anyone want to fix the oldest open spack issue? 🤣

**Comment by becker33** [comment]
Yep I'm fixing it along the lines of what I proposed
